### DISCUSSION TITLE
What will the branch in the repository when contributing?

### DISCUSSION THREAD

**Discussion by PedroSoaresDias** [vercel/next.js#64519] [Help]
### Summary

I'm with question about project branches, how can I make my branch without compromising the project, is there any pattern for creating branches for this repository?

**Comment by Icegreeen** [comment]
> ### Summary
> I'm with question about project branches, how can I make my branch without compromising the project, is there any pattern for creating branches for this repository?
> 
> ### Additional information
> _No response_
> 
> ### Example
> _No response_

To contribute to Next.js, it's recommended to follow a standard branching workflow, such as the `Gitflow model`. This involves creating a new branch from the main branch (usually "main" or "master"), naming it descriptively for your contribution (for example, feature/feature-name or fix/bug-name), making necessary changes on that branch, and then creating a pull request to merge your changes back into the main project branch. Be sure to review the project's contribution guidelines, if any, to better understand the specific contribution process.

Follow the links to supporting material.t:

1. Next.js Contribution Guide: [Next.js Contribution Guide](https://nextjs.org/docs/community/contribution-guide)

2. How to Contribute to Open Source: [Opening a Pull Request](https://opensource.guide/how-to-contribute/#opening-a-pull-request)

**Reply by PedroSoaresDias** [reply]
I will look carefully at the ways of contributing

**Comment by icyJoseph** [comment]
The main branch in Next.js is `canary`, and you start your branch from there. More over, you should first fork the project into your account, and then create a new branch in your forked version. 

Then you add the original Next.js as upstream, https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/configuring-a-remote-repository-for-a-fork.

Do you work and push to GitHub, then you can make a PR from your forked project's branch to Next.js - Fill in the form and such that the PR template presents, and wait.

As much as possible, try to create PR's that solve actual issues, rather than spontaneously opening PRs.

**Reply by PedroSoaresDias** [reply]
If I create an example project for Next.JS in the examples folder, which could be a Next.JS project using some Framework or library, is there any chance that the PR will be valid?

**Reply by icyJoseph** [reply]
The PR template has a section for that I believe. The people responsible for examples will take a look and make a decision.

**Reply by PedroSoaresDias** [reply]
If you use any Framework or library in the example project inside the examples folder, the project name would be like this: "with-(name of the Framework or library / some technology)"?

**Comment by PedroSoaresDias** [comment]
About the created branch names, I can create a branch with any name or follow pattern for example (example, feature/feature-name, or fix/bug-name)?
I would to contribute to Next.JS created an example project with Bootstrap Framework to start and later contribute to outher form.
Thank you for help me.

**Reply by GabenGar** [reply]
The branch names don't matter since they get fed into `canary` anyway.

**Reply by PedroSoaresDias** [reply]
The branch name don't matter if the merge to branch `canary` for example: `Username/branch` to `vercel/canary`?
